tools/model-compiler/src/main/java/io/spine/tools/compiler/descriptor/FileDescriptorSuperset.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.compiler.descriptor;
+
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import com.google.protobuf.DescriptorProtos.FileDescriptorSet;
+import io.spine.code.proto.FileDescriptorSets;
+import io.spine.logging.Logging;
+import io.spine.tools.compiler.archive.ArchiveEntry;
+import io.spine.tools.compiler.archive.ArchiveFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Lists.newLinkedList;
+import static io.spine.code.proto.FileDescriptors.KNOWN_TYPES;
+import static io.spine.tools.compiler.archive.ArchiveFile.isArchive;
+import static io.spine.util.Exceptions.illegalStateWithCauseOf;
+import static io.spine.util.Exceptions.newIllegalStateException;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * A set of {@code FileDescriptorSet}s.
+ */
+public final class FileDescriptorSuperset implements Logging {
+
+    private final Collection<FileDescriptorSet> descriptors;
```

Comment:
It would be better to have the field as `LinkedList` and document it why we have it this way. Having it as `Collection` is misleading.

Reply:
Or, have it just as `List` with `ArrayList` implementation, if we don't need specifically `LinkedList`.

Reply:
But we operate with `Set`s in methods that collection `FileDescriptorProto`. Why `List` here?